components/dashboard/src/workspaces/Workspaces.tsx

```diff
@@ -40,6 +43,8 @@ export default function () {
         })();
     }, [teams, location]);
 
+    const hasPreferredIde = typeof user?.additionalData?.ideSettings?.defaultIde !== "undefined";
```

Comment:
Also, we can provide a interface in `User` namespace and implement it, so that others (me 🙈) can reuse the same logic.

We can add logic in `isOnboarding` if @loujaybee want to change onboarding condition
```ts
export namespace User {
    ...
    export function hasPerferredIde(user: User) {
        return !user?.additionalData?.ideSettings?.defaultIde;
    }
    export function isOnboarding(user: User) {
        return !hasPerferredIde(user);
    }
}
```

https://github.com/gitpod-io/gitpod/commit/24630fcacd7278bd1c047eb640d712a15a9065e9#diff-957d9dd15b947e3969539ebd0dabcb8b40ce8b5564e5820cd9bb27d88be1dbc7R98-R103

Reply:
done 👍